### DISCUSSION TITLE
Unable to login afer upgrade to 0.67.2

### DISCUSSION THREAD

**Discussion by OCram85** [outline/outline#4766] [Self Hosting] [answered]
Just noticed, I can't login to outline after updating to latest version.
When trying to login I got redirected to the landing page again after authetication.

Where there any changes in in OIDC workflow?

Outline log:
```
ERR Error during authentication | error=A name field was not returned in the profile parameter, but is required. stack=UnauthorizedError: A name field was not returned in the profile parameter, but is required.
    at AuthenticationError (/opt/outline/build/server/errors.js:38:34)
    at OAuth2Strategy._verify (/opt/outline/build/server/routes/auth/providers/oidc.js:63:47)
    at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:196:24
    at OAuth2Strategy._passportOauth.Strategy.userProfile (/opt/outline/build/server/routes/auth/providers/oidc.js:32:12)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

OIDC Config:

```
# Gitea Auth
      OIDC_CLIENT_ID: 12345
      OIDC_CLIENT_SECRET: gto_1234
      OIDC_AUTH_URI: https://gitea.foo.com/login/oauth/authorize
      OIDC_TOKEN_URI: https://gitea.foo.com/login/oauth/access_token
      OIDC_USERINFO_URI: https://gitea.foo.com/login/oauth/userinfo

      OIDC_USERNAME_CLAIM: preferred_username
      OIDC_DISPLAY_NAME: OpenID
      OIDC_SCOPES: openid profile email
```

Gitea log:

```
2023/01/24 19:18:51 [63d0210b] router: completed POST /user/two_factor for 89.244.191.73:0, 303 See Other in 5.5ms @ auth/2fa.go:45(auth.TwoFactorPost)
2023/01/24 19:18:51 [63d0210b-2] router: completed GET /login/oauth/authorize?response_type=code&redirect_uri=https%3A%2F%2Foutline.foo.com%2Fauth%2Foidc.callback&scope=openid%20profile%20email&state=0bbce55828b202b1&client_id=6cc369e4-9127-42ff-9450-1cb24fe4ec6b for 89.244.191.73:0, 303 See Other in 4.8ms @ auth/oauth.go:356(auth.AuthorizeOAuth)
2023/01/24 19:18:51 [63d0210b-3] router: completed POST /login/oauth/access_token for 172.18.0.1:0, 200 OK in 101.7ms @ auth/oauth.go:589(auth.AccessTokenOAuth)
2023/01/24 19:18:52 [63d0210c] router: completed GET /login/oauth/userinfo for 172.18.0.1:0, 200 OK in 6.4ms @ auth/oauth.go:271(auth.InfoOAuth)
2023/01/24 19:18:55 [63d0210f] router: completed GET /login/oauth/authorize?response_type=code&redirect_uri=https%3A%2F%2Foutline.foo.com%2Fauth%2Foidc.callback&scope=openid%20profile%20email&state=7249d463c08c8cab&client_id=6cc369e4-9127-42ff-9450-1cb24fe4ec6b for 89.244.191.73:0, 303 See Other in 5.8ms @ auth/oauth.go:356(auth.AuthorizeOAuth)
2023/01/24 19:18:55 [63d0210f-2] router: completed POST /login/oauth/access_token for 172.18.0.1:0, 200 OK in 101.5ms @ auth/oauth.go:589(auth.AccessTokenOAuth)
2023/01/24 19:18:55 [63d0210f-3] router: completed GET /login/oauth/userinfo for 172.18.0.1:0, 200 OK in 7.2ms @ auth/oauth.go:271(auth.InfoOAuth)
```

**Comment by massimiliano-dalcero** [comment, marked as answer]
I had a similar issue.
I solved setting the field "full name" in user profile in Gitea?

![image](https://user-images.githubusercontent.com/5049867/214395257-79442212-4707-4be5-be6d-e1b8b96ce8b2.png)

**Reply by tommoor** [reply]
This is right, the name field was always required but was not correctly enforced in the previous version

**Reply by OCram85** [reply]
Awesome. Thanks for the help. Is there a way to use the account name as default or backup?

**Reply by tommoor** [reply]
What do you mean by account name, `username`?

**Reply by OCram85** [reply]
Unfortunately the `Full Name` field in a Gitea account isn't mandatory. So if outline tries to resolve the username for onboarding or login a user it causes this problem. I was thinking about using  the Gitea `username` field instead of  `Full Name` for outline.

**Reply by tommoor** [reply]
Makes sense, I'm just not sure how this was working before for your team – did everyone that signed up have an empty name in Outline?

**Reply by harisshafiq08** [reply]
@tommoor I am facing a similar issue with the Mattermost OIDC, can you please provide a support for username field instead of Full Name?

**Reply by OCram85** [reply]
> Makes sense, I'm just not sure how this was working before for your team – did everyone that signed up have an empty name in Outline?

I just integrated outline into our stack. So the team who is already working with outline is fairly small. Some just registered a new Gitea account and others removed their full name filed in gitea to avoid showing their real name in public repos.

**Comment by harisshafiq08** [comment]
@tommoor  I am facing a similar issue with Mattermost Oath/OIDC. Can you please help

`{"error":"A name field was not returned in the profile parameter, but is required.","stack":"UnauthorizedError: A name field was not returned in the profile parameter, but is required.\n    at AuthenticationError (/opt/outline/build/server/errors.js:38:34)\n    at OAuth2Strategy._verify (/opt/outline/build/server/routes/auth/providers/oidc.js:63:47)\n    at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:196:24\n    at OAuth2Strategy._passportOauth.Strategy.userProfile (/opt/outline/build/server/routes/auth/providers/oidc.js:32:12)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","level":"error","message":"Error during authentication"}
`

**Comment by msklywenn** [comment]
A user ended up with this too. The error message is not very helpful in Outline. Would it be possible to make OIDC_USERNAME_CLAIM an array so outline can try fallback sources for the username ?

**Comment by jrevillard** [comment]
Hello,

I'm facing the same issue with Keycloak.... not all our users have username set. Would it be possible to specify which field we would like to use for username ? I would like to use preferred_username..

This is where is the problem for me: https://github.com/outline/outline/blob/main/plugins/oidc/server/auth/oidc.ts#L97

Best,
Jerome

**Reply by tommoor** [reply]
This is what the `OIDC_USERNAME_CLAIM` is for

**Reply by msklywenn** [reply]
It looks like OIDC_USERNAME_CLAIM is read *after* throwing the auth error

**Reply by jrevillard** [reply]
> It looks like OIDC_USERNAME_CLAIM is read _after_ throwing the auth error

exactly

I'm testing this simple patch right now on our instance: https://github.com/outline/outline/compare/main...jrevillard:outline:patch-1

**Reply by jrevillard** [reply]
My patch works.

**Reply by jrevillard** [reply]
see #5193

**Reply by tommoor** [reply]
Tweaked and landed the patch, it'll be in the next release.
